packages/open-payments/src/client/index.ts

```diff
@@ -239,16 +256,29 @@ export const createUnauthenticatedClient = async (
   }
 }
 
-export interface CreateAuthenticatedClientArgs
-  extends CreateUnauthenticatedClientArgs {
+interface BaseAuthenticatedClientArgs extends CreateUnauthenticatedClientArgs {
+  /** The wallet address which the client will identify itself by */
+  walletAddressUrl: string
+}
+
+interface PrivateKeyConfig {
   /** The private EdDSA-Ed25519 key (or the relative or absolute path to the key) with which requests will be signed */
   privateKey: string | KeyLike
   /** The key identifier referring to the private key */
   keyId: string
-  /** The wallet address which the client will identify itself by */
-  walletAddressUrl: string
 }
 
+interface InterceptorConfig {
+  /** The custom request interceptor to use. */
+  requestInterceptor: InterceptorFn
+}
+
+export type CreateAuthenticatedClientArgs = BaseAuthenticatedClientArgs &
+  PrivateKeyConfig
+
+export type CreateExperimentalAuthenticatedClientArgs =
```

Comment:
I think the only usage of "experimental" should be in the JSdoc. Otherwise, the types should just be called what they are, eg. `CreateAuthenticatedClientWithRequestInterceptorArgs` or something like that (even though it's extremely long lol)